scraper.py

```diff
@@ -0,0 +1,147 @@
+    # with open(filename, "w") as f:
+        # f.write(content)
+    file = open(filename, "w")
+    file.write(str(content))
+
+
+def load_inspection_page(name, **kwargs):
+    content, encoding = get_inspection_page(**kwargs)
+    write_file(content.decode(encoding), name)
+    file = io.open(name, "r")
+    return file
+
+
+def parse_source(html):
+    parsed = BeautifulSoup(html, 'html5lib')
+    return parsed
+
+
+def extract_data_listings(html):
+    return html.findAll('div', {"id": lambda x: x and x.startswith('PR') and x.endswith("~")})
+
+    # id_finder = re.compile(r'PR[\d]+~')
+    # return html.find_all('div', id=id_finder)
+
+
+def has_two_tds(elem):
+    is_tr = elem.name == 'tr'
+    td_children = elem.find_all('td', recursive=False)
+    has_two = len(td_children) == 2
+    return is_tr and has_two
+
+
+def clean_data(td):
+    data = td.string
+    try:
+        return data.strip(" \n:-")
+    except AttributeError:
+        return u""
+
+
+def extract_restaurant_metadata(elem):
+    metadata_rows = elem.find('tbody').find_all(
+        has_two_tds, recursive=False
+    )
+    rdata = {}
+    current_label = ''
+    for row in metadata_rows:
+        key_cell, val_cell = row.find_all('td', recursive=False)
+        new_label = clean_data(key_cell)
+        current_label = new_label if new_label else current_label
+        rdata.setdefault(current_label, []).append(clean_data(val_cell))
+    return rdata
+
+
+def is_inspection_row(elem):
+    is_tr = elem.name == 'tr'
+    if not is_tr:
+        return False
+    td_children = elem.find_all('td', recursive=False)
+    has_four = len(td_children) == 4
+    this_text = clean_data(td_children[0]).lower()
+    contains_word = 'inspection' in this_text
+    does_not_start = not this_text.startswith('inspection')
+    return is_tr and has_four and contains_word and does_not_start
+
+
+def extract_score_data(elem):
+    inspection_rows = elem.find_all(is_inspection_row)
+    samples = len(inspection_rows)
+    total = high_score = average = 0
+    for row in inspection_rows:
+        strval = clean_data(row.find_all('td')[2])
+        try:
+            intval = int(strval)
+        except (ValueError, TypeError):
+            samples -= 1
+        else:
+            total += intval
+            high_score = intval if intval > high_score else high_score
+    if samples:
+        average = total / float(samples)
+    data = {
+        u'Average Score': average,
+        u'High Score': high_score,
+        u'Total Inspections': samples
+    }
+    return data
+
+
+if __name__ == '__main__':
+    kwargs = {
+        'Inspection_Start': '2/1/2013',
+        'Inspection_End': '2/1/2015',
+        'Zip_Code': '98109'
+    }
+    if len(sys.argv) > 1 and sys.argv[1] == 'test':
+        html, encoding = load_inspection_page('inspection_page.html')
+    else:
+        html, encoding = get_inspection_page(**kwargs)
+    doc = parse_source(html)
+    write_file(doc, 'inspection_page.html')
```

Comment:
You probably don't want to do this every time.  If you're testing and reading from `inspection_page.html` already, then there's certainly no reason to write back out to it.